### ISSUE TITLE
Temperature calibration, as supported in the runtime 

### ISSUE THREAD

**Issue by SolarPolarMan** [bbcmicrobit/micropython#413]
Could we see support for temperature calibration [since it's in the runtime](https://lancaster-university.github.io/microbit-docs/ubit/thermometer/#int-setcalibration-int-calibrationtemp)? If not, could you shed some light on how the temperature calibration function works so I can imitate it?

**Comment by dpgeorge** [comment]
> If not, could you shed some light on how the temperature calibration function works so I can imitate it?

I think it's simply taking the difference between the measured and actual value, then adding that to subsequent measurements.  Eg:
```python
def calibrate_temp(t):
    offset = t - temperature()
def get_calibrated_temp():
    return temperature() + offset
```

**Comment by SolarPolarMan** [comment]
So we just assume that the chip stays at the same temperature, relative to the environment?

**Comment by dpgeorge** [comment]
> So we just assume that the chip stays at the same temperature, relative to the environment?

No, the chip changes temperature as the environment changes.  It should be quite accurate in terms of relative temperature change, but not the absolute value.  Hence given 1 data point (the calibration value) you should be able to compute an accurate absolute temperature.
